apps/extension/src/App/App.tsx

```diff
@@ -1,57 +1,88 @@
 import React, { useEffect, useState } from "react";
-import { BrowserRouter, Routes, Route } from "react-router-dom";
 import { ThemeProvider } from "styled-components";
-import browser from "webextension-polyfill";
+import { Routes, Route, useNavigate } from "react-router-dom";
 
-import { Button, ButtonVariant } from "@anoma/components";
 import { getTheme } from "@anoma/utils";
-
 import { ExtensionRequester } from "extension";
 import { Ports } from "router";
-import { DerivedAccount, QueryAccountsMsg } from "background/keyring";
+import {
+  CheckIsLockedMsg,
+  DerivedAccount,
+  QueryAccountsMsg,
+} from "background/keyring";
+
 import {
   AppContainer,
   BottomSection,
   ContentContainer,
-  LoadingError,
   GlobalStyles,
   TopSection,
 } from "./App.components";
-import Accounts from "./Accounts/Accounts";
-import AddAccount from "./Accounts/AddAccount";
-import { TopLevelRoute } from "App/types";
+import { LockWrapper } from "./LockWrapper";
+import { Accounts, AddAccount } from "./Accounts";
+import { Login } from "./Login";
+import { Setup } from "./Setup";
+import { TopLevelRoute } from "./types";
+import { Loading } from "./Loading";
 
 const requester = new ExtensionRequester();
 
-enum Status {
+export enum Status {
   Completed,
   Pending,
   Failed,
 }
 
 export const App: React.FC = () => {
   const theme = getTheme(true, false);
+  const navigate = useNavigate();
   const [status, setStatus] = useState<Status>();
   const [accounts, setAccounts] = useState<DerivedAccount[]>([]);
+  const [error, setError] = useState("");
 
-  useEffect(() => {
+  const fetchAccounts = async (): Promise<void> => {
     setStatus(Status.Pending);
-    (async () => {
-      try {
-        const accounts = await requester.sendMessage(
-          Ports.Background,
-          new QueryAccountsMsg()
-        );
-        setAccounts(accounts);
-      } catch (e) {
-        console.error(e);
-        setStatus(Status.Failed);
-      } finally {
-        setStatus(Status.Completed);
-      }
-    })();
+    try {
+      const accounts = await requester.sendMessage(
+        Ports.Background,
+        new QueryAccountsMsg()
+      );
+      setAccounts(accounts);
+    } catch (e) {
+      console.error(e);
+      setError(`An error occurred while loading extension: ${e}`);
+      setStatus(Status.Failed);
+    } finally {
+      setStatus(Status.Completed);
+    }
+  };
+
+  const checkIsLocked = async (): Promise<void> => {
```

Comment:
nit: the naming makes it feel as if it checks if something is locked, while it also navigates.